src/rules/at-rule-blockless-newline-after/index.js

```diff
@@ -0,0 +1,60 @@
+import {
+  cssStatementHasBlock,
+  rawNodeString,
+  report,
+  ruleMessages,
+  validateOptions,
+  whitespaceChecker,
+} from "../../utils"
+
+export const ruleName = "at-rule-blockless-newline-after"
+
+export const messages = ruleMessages(ruleName, {
+  expectedAfter: () => "Expected newline after blockless at-rule",
+})
+
+export default function (actual) {
+  const checker = whitespaceChecker("newline", actual, messages)
+
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual,
+      possible: ["always"],
+    })
+    if (!validOptions) { return }
+
+    root.walkAtRules(statement => {
+
+      const isNested = statement.parent !== root
```

Comment:
This was intentional, although I'd be happy to remove it. My thinking behind the nested check was that all the blockless rules (`import`, `charset` and `namespace`) are ignored when nested. Seemingly because each of them is a special case, rather than that being a feature of all blockless at-rules.

https://www.w3.org/TR/CSS2/cascade.html#at-import
https://www.w3.org/TR/css-syntax-3/#charset-rule
https://www.w3.org/TR/css3-namespace/#syntax